### PR TITLE
fix(thumbnail):fixed assertions for thumbnail operation

### PR DESCRIPTION
**Pull request by cnlangzi**
*changes*
- improved image file generator 
- fixed all assertions for thumbnail

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/fix_thumbnail_assertion  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | fix/update_sha1_with_sha256_for_file_hash  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/update_sha1_with_sha256_for_file_hash  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/fix_thumbnail_assertion  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | fix/update_sha1_with_sha256_for_file_hash  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/update_sha1_with_sha256_for_file_hash  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | fix/fix_thumbnail_assertion  |
| gosdk  | fix/update_sha1_with_sha256_for_file_hash  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | fix/update_sha1_with_sha256_for_file_hash  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

File: tests/cli_tests/flaky___broken_scenarios_test.go [hunk 1]
```diff
@@ -539,12 +540,14 @@ func Test___FlakyBrokenScenarios(t *testing.T) {
...
 
+		err := os.WriteFile(thumbnail, thumbnailBytes, os.ModePerm)
+		require.Nil(t, err, "failed to generate thumbnail", err)
 		localFilePath := generateFileAndUploadWithParam(t, allocationID, remotepath, filesize, map[string]interface{}{"thumbnailpath": thumbnail})
 
-		output, err = downloadFile(t, configPath, createParams(map[string]interface{}{
+		output, err := downloadFile(t, configPath, createParams(map[string]interface{}{
 			"allocation": allocationID,
 			"remotepath": remotepath + filepath.Base(localFilePath),
 			"localpath":  "tmp/",
```
Comment (Kishan-Dhakan) [inline tests/cli_tests/flaky___broken_scenarios_test.go:553]: If this passes consistently now, it should be moved to `zboxcli_file_update_test.go`.

File: tests/cli_tests/zboxcli_download_test.go [hunk 2]
```diff
@@ -568,15 +568,20 @@ func TestDownload(t *testing.T) {
 		err = os.Remove(filename)
 		require.Nil(t, err)
 
+		localPath := "tmp/" + filepath.Base(filename)
+
```
Comment (Kishan-Dhakan) [inline tests/cli_tests/zboxcli_download_test.go:572]: Can this be changed to `os.TempDir()`?

Comment (cnlangzi) [reply tests/cli_tests/zboxcli_download_test.go:572]: it seems is used everywhere to write file inside work dir. should we change it with os.TempDir() ?

**Review by stewartie4 — COMMENTED** [review]
LGTM

**Review by stewartie4 — APPROVED** [review]

**Review by kushthedude — APPROVED** [review]

**Review by Kishan-Dhakan — APPROVED** [review]

**Pull request merged**